src/hooks/v2v3/V2V3ProjectContracts.ts

```diff
@@ -0,0 +1,33 @@
+import { Contract } from '@ethersproject/contracts'
+import { readProvider } from 'constants/readProvider'
+import { V2V3ContractsContext } from 'contexts/v2v3/V2V3ContractsContext'
+import { useContext } from 'react'
+import useProjectController from './contractReader/ProjectController'
+
+/**
+ * Load project-specific JB contracts.
+ * @returns
+ */
+export function useV2V3ProjectContracts({ projectId }: { projectId: number }) {
+  const { contracts } = useContext(V2V3ContractsContext)
+
+  /**
+   * Load controller data
+   */
+  const { data: controllerAddress } = useProjectController({
+    projectId,
+  })
+
+  const JBController =
+    controllerAddress && contracts
+      ? new Contract(
+          controllerAddress,
+          contracts.JBController.interface,
```

Comment:
do we care that the V2 & V3 interfaces will be a bit different?

Reply:
I think they're the same, no?

Reply:
I did a quick diff comparison and it seems they're different, maybe I used the wrong files. I'll check again when back at my laptop

Reply:
eh. I was going off this but didnt check myself 😅  https://discord.com/channels/939317843059679252/1018993050900512790/1020027898603774023

Lemme do a real check.

Reply:
ah shit, you're right they are a bit different. Fixable though - lemme see.

Reply:
Done @peripheralist , lmkwyt.

https://github.com/jbx-protocol/juice-interface/pull/2116/commits/80390702be3b902b1e349c090ad1a65300942b73#diff-94744575e283398812b18d59783aaed37032dc53d3e0c6e27373025b4a9a188d